maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -179,28 +184,52 @@ public fun GoogleMap(
         }
     }
 
+    val lifecycleOwner = LocalLifecycleOwner.current
     var isCompositionSet by remember { mutableStateOf(false) }
     val mapUpdaterScope = rememberCoroutineScope()
 
     AndroidView(
         modifier = modifier,
         factory = {
-            Log.d(TAG, "Factory")
             debugIsMapReused = false
             MapView(context, googleMapOptionsFactory()).also { mapView ->
+                mapView.log("Creating MapView")
                 mapView.registerAndSaveNewComponentCallbacks(context)
+
+                // Used to observe lifecycle owner's lifecycle state and
+                // to gradually move between states.
+                mapView.tagData().lifecycleRegistry = DerivedLifecycleRegistry(lifecycleOwner) { _, event ->
+                    mapView.log("Invoking $event")
+                    when (event) {
+                        Lifecycle.Event.ON_CREATE -> mapView.onCreate(Bundle())
+                        Lifecycle.Event.ON_START -> mapView.onStart()
+                        Lifecycle.Event.ON_RESUME -> mapView.onResume()
+                        Lifecycle.Event.ON_PAUSE -> mapView.onPause()
+                        Lifecycle.Event.ON_STOP -> mapView.onStop()
+                        Lifecycle.Event.ON_DESTROY -> {
+                            mapView.onDestroy()
+                            mapView.tagData().lifecycleRegistry!!.destroy()
+                        }
+                        else -> error("Unsupported Lifecycle.Event: $event")
+                    }
+                }
             }
         },
         onReset = { mapView ->
             // View is detached.
+            mapView.tagData().lifecycleRegistry!!.overrideLifecycleState(Lifecycle.State.CREATED)
```

Comment:
Currently the `LifecycleObserver` is added before the `MapView` has been attached.
This will result in onCreate->onStart->onResume being invoked before it's attached.
Are you saying this is a bad thing? Is that why you wanted to add the `LifecycleObserver` only in `onViewAttachedToWindow`?
I don't see anything in the documentation about that the `MapView` has to be attached before invoking the lifecycle events.
Maybe it's just a bad practice.